hs_communities/views/communities.py

```diff
@@ -0,0 +1,138 @@
+class MyCommunitiesView(TemplateView):
+    template_name = 'hs_communities/my-communities.html'
+
+    def dispatch(self, *args, **kwargs):
+        return super(MyCommunitiesView, self).dispatch(*args, **kwargs)
+
+    def group_to_community(self, grp, communities):
+        """
+        return the community membership information of a group; group can belong to only one community
+        :param grp: group object
+        :param communities: communities
+        :return: tuple id, name of community
+        """
+        for community in communities:
+            if grp.id in [g.id for g in community.member_groups]:
+                return community
+
+    def get_context_data(self, **kwargs):
+        grps_member_of = []
+        u = User.objects.get(pk=self.request.user.id)
+        groups = Group.objects.filter(gaccess__active=True).exclude(name="Hydroshare Author")
+        # for each group set group dynamic attributes
+        for g in groups:
+            g.is_user_member = u in g.gaccess.members
+            if g.is_user_member:
+                grps_member_of.append(g)
+            g.join_request_waiting_owner_action = g.gaccess.group_membership_requests.filter(request_from=u).exists()
+            g.join_request_waiting_user_action = g.gaccess.group_membership_requests.filter(invitation_to=u).exists()
+            g.join_request = None
+            if g.join_request_waiting_owner_action or g.join_request_waiting_user_action:
+                g.join_request = g.gaccess.group_membership_requests.filter(request_from=u).first() or \
+                                 g.gaccess.group_membership_requests.filter(invitation_to=u).first()
+
+        comms_member_of = [self.group_to_community(g, Community.objects.all()) for g in grps_member_of]
+        return {
+            'communities_list': [c for c in comms_member_of if c is not None]
+        }
+
+
+@method_decorator(login_required, name='dispatch')
+class TopicsView(TemplateView):
+    """
+    action: CREATE, READ, UPDATE, DELETE
+    """
+
+    def get(self, request, *args, **kwargs):
+        return render(request, 'pages/topics.html', {'topics_json': self.get_topics_data()})
+
+    def post(self, request, *args, **kwargs):
+        if request.POST.get('action') == 'CREATE':
+            try:
+                new_topic = Topic()
+                new_topic.name = request.POST.get('name').replace("--", "")
+                new_topic.save()
+            except Exception as e:
+                print("TopicsView error creating new topic {}".format(e))
+        elif request.POST.get('action') == 'UPDATE':
+            try:
+                update_topic = Topic.objects.get(id=request.POST.get('id'))
+                update_topic.name = request.POST.get('name')
+                update_topic.save()
+            except Exception as e:
+                print("TopicsView error updating topic {}".format(e))
+        elif request.POST.get('action') == 'DELETE':
```

Comment:
Why not use delete instead of post for this?

Reply:
PUT and DELETE have a different implementation in this version of Django, in my understanding. That is why the additional ('action') parameter was created.

Reply:
Pretty sure you can write any of the REST methods with templateview in 1.11.  https://docs.djangoproject.com/en/1.11/ref/class-based-views/base/#django.views.generic.base.View.http_method_names